File: Vanilla/sources/zdoc/media/lua/shared/Library/ItemRecipe.lua
---@class ItemRecipe : zombie.scripting.objects.ItemRecipe
---@field public name String
---@field public use Integer
---@field public cooked Boolean
---@field private module String
ItemRecipe = {}

---@public
---@return String
function ItemRecipe:getModule() end

---@public
---@return String
function ItemRecipe:getName() end

---@public
---@param arg0 String
---@return void
function ItemRecipe:setModule(arg0) end

---@public
---@return Integer
function ItemRecipe:getUse() end

---@public
---@return String
function ItemRecipe:getFullType() end

File: Vanilla/sources/zomboid/zombie/scripting/objects/ItemRecipe.java
package zombie.scripting.objects;

public final class ItemRecipe {
   public String name;
   public Integer use = -1;
   public Boolean cooked = false;
   private String module = null;

   public Integer getUse() {
      return this.use;
   }

   public ItemRecipe(String var1, String var2, Integer var3) {
      this.name = var1;
      this.use = var3;
      this.setModule(var2);
   }

   public String getName() {
      return this.name;
   }

   public String getModule() {
      return this.module;
   }

   public void setModule(String var1) {
      this.module = var1;
   }

   public String getFullType() {
      return this.module + "." + this.name;
   }
}
